src/title/Title.tsx

```diff
@@ -1,25 +1,32 @@
-import React, { Fragment, ReactNode } from 'react'
+import React, { ReactNode } from 'react'
 import cc from 'classcat'
 
+import { StyledTitle } from './Title.style'
+
 export interface TitleProps {
   readonly id?: string
   readonly className?: string
   readonly children: ReactNode
-  readonly headingLevel?: number | string
+  readonly headingLevel?: 1 | 2 | 3 | 4 | 5 | 6 | '1' | '2' | '3' | '4' | '5' | '6'
 }
 
-export const isHeadingAvailable = (level: number) => level >= 1 && level <= 6
+const HEADING_LEVEL_MAPPING = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const
 
 export const Title = ({ id, className, children, headingLevel = 1 }: TitleProps) => {
-  if (!isHeadingAvailable(Number(headingLevel))) {
-    return null
+  let headingLevelNumber: number
+  if (typeof headingLevel === 'string') {
+    headingLevelNumber = parseInt(headingLevel, 10)
+  } else {
+    headingLevelNumber = headingLevel
   }
-  return React.createElement(
-    `h${headingLevel}`,
-    {
-      id,
-      className: cc(['kirk-title', className]),
-    },
-    <Fragment>{children}</Fragment>,
+
+  return (
+    <StyledTitle
+      as={HEADING_LEVEL_MAPPING[headingLevelNumber - 1]}
```

Comment:
you could simplify by interpolating `h$headingLevel`

Reply:
Unfortunately interpolating is not working as expected with the as polymorphic and this workaround is needed.
Happy to give you more details on this where @simonrelet helped me (thanks!)

Reply:
``` `h${headingLevel}` ``` cannot be typed as `'h1' | 'h2' | ... | 'h6'`, it has to be explicitly `'h1'`, `'h2'`, ..., `'h6'`.

@emaghe, I would add a comment explaining the reason of this mapping.